internal/webhooks/patch_validation.go

```diff
@@ -165,10 +165,9 @@ func validateEnabledIf(enabledIf *string, path *field.Path) field.ErrorList {
 func validateSelectors(selector clusterv1.PatchSelector, class *clusterv1.ClusterClass, path *field.Path) field.ErrorList {
 	var allErrs field.ErrorList
 
-	// Return an error if none of the possible selectors are enabled.
-	if !(selector.MatchResources.InfrastructureCluster || selector.MatchResources.ControlPlane ||
-		(selector.MatchResources.MachineDeploymentClass != nil && len(selector.MatchResources.MachineDeploymentClass.Names) > 0) ||
-		(selector.MatchResources.MachinePoolClass != nil && len(selector.MatchResources.MachinePoolClass.Names) > 0)) {
+	if !selector.MatchResources.InfrastructureCluster && !selector.MatchResources.ControlPlane &&
```

Comment:
```suggestion
	// Return an error if none of the possible selectors are enabled.
	if !selector.MatchResources.InfrastructureCluster && !selector.MatchResources.ControlPlane &&
```
Not sure why the comment was dropped, let's keep it

Reply:
I think this comment is still missing